trillian_map_api.proto

```diff
@@ -58,7 +59,15 @@ message GetMapLeavesResponse {
 message SetMapLeavesRequest {
   int64 map_id = 1;
   repeated MapLeaf leaves = 2;
-  MapperMetadata mapper_data = 3;
+  reserved 3;  // was MapperMetadata (removed, replaced by mapper_metadata).
+  // Metadata that the Map should associate with the new Map root after
```

Comment:
There's an assumption of a 1:1 relationship between SetMapLeaves calls and SignedMapRoots. Is that true/a guarantee we'd like to make? If not, what happens in regards to the metadata?

Reply:
I believe the Trillian Map generates a new signed map root after every call to SetLeaves.
However this is not documented, and feels to me like something that might end up having to be made tweakable in a practical deployment?

Reply:
If it's tweakable, then what happens to the metadata when the 1:1 relationship is broken? Does it get overwritten? Do we need a `repeated metadata` in SignedMapRoot?

Reply:
I have misled you :).  It isn't tweakable.  As I understand it there is no sequencing in the Map, so as soon as a SetLeaves RPC has completed there will have been a new signed map head created, which is returned in the SetLeaves response.  So there is a 1:1 relationship, although it isn't explicitly stated in the API.

My intended meaning was to express some doubt as to whether this will be performant for a map under heavy load, but I guess we cross that bridge when we get to it.

Reply:
👍 it's intended to both enable mapper personalities to 'pick up' where they left off, as well as provide  a commitment from the system that the given root hash is reproducible at the associated mapper state. The latter allows monitors to verify the correct operation of the mapper & Map.

Reply:
So do you think a verifying monitor (monitoring the mapper/Map) be likely to iterate through the historic SignedMapHeads and use the metadata at each snapshot to conduct its verification?  Or is this getting a bit application-specific?

Reply:
I do think the 'verify that a given root hash is reproducible at the associated mapper state' is a rather application (personality)-specific thing.  It assumes that the Map is being used in conjunction with an append-only log, and that the full log can be queried by the verifying party.  That might not always be the case?